streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollector.java

```
@@ -64,9 +68,20 @@ public RecordCollector(Producer<byte[], byte[]> producer) {
     }
 
     public <K, V> void send(ProducerRecord<K, V> record, Serializer<K> keySerializer, Serializer<V> valueSerializer) {
+        send(record, keySerializer, valueSerializer, null);
+    }
+
+    public <K, V> void send(ProducerRecord<K, V> record, Serializer<K> keySerializer, Serializer<V> valueSerializer,
+                            StreamPartitioner<K, V> partitioner) {
         byte[] keyBytes = keySerializer.serialize(record.topic(), record.key());
         byte[] valBytes = valueSerializer.serialize(record.topic(), record.value());
-        this.producer.send(new ProducerRecord<>(record.topic(), keyBytes, valBytes), callback);
+        Integer partition = null;
+        if (partitioner != null) {
+            List<PartitionInfo> partitions = this.producer.partitionsFor(record.topic());
+            if (partitions != null)
+                partition = partitioner.partition(record.key(), record.value(), partitions.size());
```

If ```partitions``` does not exist, should we throw an exception?